src/controllers/user-data-manager.js

```diff
@@ -0,0 +1,132 @@
+import WordItemIndexedDbDriver from '@/storage/worditem-indexeddb-driver.js'
+import WordItemRemoteDbDriver from '@/storage/worditem-remotedb-driver.js'
+import IndexedDBAdapter from '@/storage/indexed-db-adapter.js'
+import RemoteDBAdapter from '@/storage/remote-db-adapter.js'
+export default class UserDataManager {
+
+  constructor (userID,events) {
+    this.userID = userID
+    events.WORDITEM_UPDATED.sub(this.update.bind(this))
+    events.WORDITEM_DELETED.sub(this.delete.bind(this))
+    events.WORDLIST_DELETED.sub(this.deleteMany.bind(this))
+  }
+
+  _localStorageAdapter(dataType) {
+    let dbDriver = new UserDataManager.LOCAL_DRIVER_CLASSES[dataType](this.userID)
```

Comment:
this all could/should be done better. It would be nice if there was a way for a class to register itself with the UserDataManager, but that might be overkill for this release.